docs/cloud-tutorials/best-practices.md

```diff
@@ -0,0 +1,87 @@
+---
+sidebar_position: 8
+title: Best practices
+description: Tips and tricks for succesfully using DBOS Cloud
+---
+
+This section dives into specific behaviors and corner cases you might encounter when using DBOS Cloud.
+
+## Managing applications
+
+### Registration
+* An application must be registered before being deployed. DBOS Cloud CLI will handle registration by default when you deploy an application for the first time.
+* Only one request at a time can manipulate an application and you will get an `application is busy` error if you attempt multiple concurrent modifications.
+* Application names are unique within an [organization](#organizations-and-users-management)
+
+### Deployment
+* Interrupting a deployment request - for example, pressing control + C - will *not* cancel the deploy action on DBOS Cloud
+* DBOS Cloud will prune all development dependencies. Make sure to explicitly install any runtime dependency you need with `npm install`
+* Your application must not use more than 1GB of disk space
+* DBOS Cloud stores each version of your application. You can download a specific version using the DBOS Cloud CLI [TOLINK WHEN MERGED]
+
+### During runtime
+* Nothing on the filesystem is persisted between deployments
+* DBOS Cloud will keep around VMs running old versions of a workflow. This means, you need to be mindful of non-backward compatible database schema changes.
+
+### Deletion
+* Deleting an app is an asynchronous process and there is a small delay between a deletion request completing and the application being fully deleted. This means you could hit an "application is busy" error when re-deploying an application you just deleted
+* An application will be deleted once all PENDING workflows -- including these previous versions -- are completed
+* All application archives -- including previous versions -- will be deleted
+* You can elect to drop the application database alongside deleting an application. This will also delete all the database roles associated with the application in the database instance.
```

Comment:
The app role is always deleted when you delete an app. Also, I don't think we need to mention it at all as the role is transparent to users.

Reply:
I wanted to mention it because they'll see the roles if they query `pg_roles`. I don't feel strongly about it, though.